run.ps1

```diff
@@ -0,0 +1,34 @@
+# set the base tools directory
+$toolsLocalPath = Join-Path (Get-Location) "Tools"
+$runExe = Join-Path $toolsLocalPath "Microsoft.DotNet.BuildTools.Run\run.exe"
+
+# if the tool doesn't exist then download and run the boot-strapper script
+if ((Test-Path $runExe) -eq 0)
+{
+    if ((Test-Path $toolsLocalPath) -eq 0)
+    {
+        mkdir $toolsLocalPath | Out-Null
+    }
+
+    # download boot-strapper script
+    $bootStrapperPath = Join-Path $toolsLocalPath "bootstrap.ps1"
+    Invoke-WebRequest "https://raw.githubusercontent.com/dotnet/buildtools/master/bootstrap/bootstrap.ps1" -OutFile $bootStrapperPath
+
+    # now execute it
+    Invoke-Expression "$bootStrapperPath (Get-Location) $toolsLocalPath" | Out-File (Join-Path (Get-Location) "bootstrap.log")
+    if ($LastExitCode -ne 0)
+    {
+        Write-Output "Boot-strapping failed with exit code $LastExitCode"
+        exit $LastExitCode
+    }
+
+    # copy nuget.exe into the packages directory (this is unique to the build tools repo)
+    $packagesPath = Join-Path (Get-Location) "packages"
+    Get-ChildItem (Join-Path $packagesPath "NuGet.CommandLine") -filter "nuget.exe" -recurse | foreach { copy $_.FullName $packagesPath }
```

Comment:
I do like the value of having a copy-paste script that is the same for all the repo's, and so I would be in favor of something like this, but before we even go there I would like to see if there is some pattern we can use in the bootstrap.ps1 or .toolsversion which would allow for this instead of needing to write this custom code?